src/reporting/Interfaces.sol

```diff
@@ -27,19 +27,28 @@ contract IMarket is Typed {
     function canBeReportedOn() constant returns (bool);
     function getFinalWinningReportingToken() constant returns (ReportingToken);
     function getFinalPayoutDistributionHash() constant returns (bytes32);
-    function derivePayoutDistributionHash(int256[]) constant returns (bytes32);
+    function derivePayoutDistributionHash(uint256[]) constant returns (bytes32);
     function updateTentativeWinningPayoutDistributionHash(bytes32) public returns (bool);
     function isInAllReportingPhase() constant public returns (bool);
     function isInLimitedReportingPhase() constant public returns (bool);
     function isDoneWithAllReporters() constant public returns (bool);
     function isDoneWithLimitedReporters() constant public returns (bool);
     function getReportingTokenOrZeroByPayoutDistributionHash(bytes32) constant public returns (ReportingToken);
+    function getFinalizationTime() constant public returns (uint256);
+    function getDenominationToken() constant public returns (ERC20);
+    function getPayoutDenominator() constant public returns (uint256);
+    function getShareToken(uint8) constant public returns (IShareToken);
+    function getCompleteSetCostInAttotokens() constant public returns(uint256);
+    function getMarketCreatorSettlementFeeInAttoethPerEth() constant public returns (uint256);
+    function getCreator() constant public returns (address);
+    function shouldCollectReportingFees() constant public returns (bool);
 }
 
 
-contract IShareToken is Typed {
+contract IShareToken is ERC20, Typed {
```

Comment:
Note to future us: We should change `ShareToken` to `VariableSupplyToken` when we port it to Solidity.